### PR TITLE
Convert group function results to PercentType if DecimalType is not s…

### LINKED ISSUE #3734: Change in Tradfri Group updates single Items, but change in Items doesn't update the Group
I have a Tradfri lamp with 5 lightbulbs.
In my items file I have the 5 brightness and color channels and two according groups (see below). When I change the group value, the single lightbulbs are all updated. But when I change the lightbulbs (via the Remote or via the Web-UI), the group value doesn't reflect those changes.

My items file:
```
Group:Dimmer:AVG wz_it_lmp1_br "Helligkeit" <dimmablelight>

Dimmer wz_it_lmp1_glb1_br "Helligkeit" <dimmablelight> (wz_it_lmp1_br) {channel="tradfri:0220:gwb072bf279bad:65537:brightness"}
Dimmer wz_it_lmp1_glb2_br "Helligkeit" <dimmablelight> (wz_it_lmp1_br) {channel="tradfri:0220:gwb072bf279bad:65538:brightness"}
Dimmer wz_it_lmp1_glb3_br "Helligkeit" <dimmablelight> (wz_it_lmp1_br) {channel="tradfri:0220:gwb072bf279bad:65539:brightness"}
Dimmer wz_it_lmp1_glb4_br "Helligkeit" <dimmablelight> (wz_it_lmp1_br) {channel="tradfri:0220:gwb072bf279bad:65540:brightness"}
Dimmer wz_it_lmp1_glb5_br "Helligkeit" <dimmablelight> (wz_it_lmp1_br) {channel="tradfri:0220:gwb072bf279bad:65541:brightness"}
```

Log when changing the value of the group:
```
2017-06-23 21:58:38.542 [ItemCommandEvent          ] - Item 'wz_it_lmp1_br' received command 26
2017-06-23 21:58:38.550 [ItemCommandEvent          ] - Item 'wz_it_lmp1_glb5_br' received command 26
2017-06-23 21:58:38.562 [ItemStateChangedEvent     ] - wz_it_lmp1_glb5_br changed from 82 to 26
2017-06-23 21:58:38.594 [ItemStateChangedEvent     ] - wz_it_lmp1_glb4_br changed from 82 to 26
2017-06-23 21:58:38.598 [ItemCommandEvent          ] - Item 'wz_it_lmp1_glb4_br' received command 26
2017-06-23 21:58:38.622 [ItemCommandEvent          ] - Item 'wz_it_lmp1_glb2_br' received command 26
2017-06-23 21:58:38.635 [ItemCommandEvent          ] - Item 'wz_it_lmp1_glb3_br' received command 26
2017-06-23 21:58:38.667 [ItemStateChangedEvent     ] - wz_it_lmp1_glb2_br changed from 82 to 26
2017-06-23 21:58:38.672 [ItemCommandEvent          ] - Item 'wz_it_lmp1_glb1_br' received command 26
2017-06-23 21:58:38.704 [ItemStateChangedEvent     ] - wz_it_lmp1_glb3_br changed from 82 to 26
2017-06-23 21:58:38.707 [ItemStateChangedEvent     ] - wz_it_lmp1_glb1_br changed from 82 to 26
```

Log when changing the value of an item:
```
2017-06-23 21:59:42.320 [ItemCommandEvent          ] - Item 'wz_it_lmp1_glb3_br' received command 100
2017-06-23 21:59:42.326 [ItemStateChangedEvent     ] - wz_it_lmp1_glb3_br changed from 26 to 100
```

### PR DESCRIPTION
**Pull request by triller-telekom**
…upported

Fixes #3734

And converts the test cases from Groovy to Java

Signed-off-by: Stefan Triller <stefan.triller@telekom.de>

### CODE REVIEW

File: bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/GroupItem.java [hunk 1]
```diff
@@ -310,8 +312,15 @@ public void stateChanged(Item item, State oldState, State newState) {
     @Override
     public void stateUpdated(Item item, State state) {
         State oldState = this.state;
-        if (function != null) {
-            setState(function.calculate(members));
+        if (function != null && baseItem != null) {
+            State calculatedState = function.calculate(members);
+
+            // functions return DecimalTypes so we have to convert them if the baseItem type only supports PercentType
+            if (baseItem.getAcceptedDataTypes().contains(PercentType.class)
```
Comment (maggu2810) [inline bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/GroupItem.java]: Wouldn't it make sense to check the type of `calculatedState` first?

```Java
if (calculatedState instanceof DecimalType && !baseItem.getAcceptedDataTypes().contains(DecimalType.class) ...)
```

File: bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/GroupItem.java [hunk 2]
```diff
@@ -310,8 +312,15 @@ public void stateChanged(Item item, State oldState, State newState) {
     @Override
     public void stateUpdated(Item item, State state) {
         State oldState = this.state;
-        if (function != null) {
-            setState(function.calculate(members));
+        if (function != null && baseItem != null) {
+            State calculatedState = function.calculate(members);
+
+            // functions return DecimalTypes so we have to convert them if the baseItem type only supports PercentType
```
Comment (sjka) [inline bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/GroupItem.java]: This seems to be a pretty specific solution... 
Is there anything speaking against using `ItemUtil.convertToAcceptedState(State, Item)` here?

**Comment by triller-telekom** [comment]
I have included the converted test in this PR because it adds a new test for the implemented fix, see [this test method](https://github.com/eclipse/smarthome/pull/3758/files#diff-670314dbd94fe4ca2d07ee8c20d6a28eR354)

File: bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/GroupItem.java (hunk 2 above)
Comment (triller-telekom) [inline bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/GroupItem.java]: Good idea, this method is exactly what i was looking for, but I was not aware of that it exists ;) I have changed it now.

**Comment by maggu2810** [comment]
> @maggu2810 okay for you too?

Okay for me. Personally I would use a one-liner and drop the variable declarations, but all fine...

**Pull request merged**